src/app/features/tech-record/tech-record-routing.module.ts

```diff
@@ -1,18 +1,21 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 import { MsalGuard } from '@azure/msal-angular';
+import { TechRecordViewResolver } from 'src/app/resolvers/tech-record-view/tech-record-view.resolver';
 import { TechRecordComponent } from './tech-record.component';
 
 const routes: Routes = [
   {
-    path: ':vin',
+    path: ':systemNumber',
     component: TechRecordComponent,
-    canActivateChild: [MsalGuard]
+    canActivateChild: [MsalGuard],
+    resolve: { load: TechRecordViewResolver }
   },
   {
-    path: ':vin/:techCreatedAt',
+    path: ':systemNumber/:techCreatedAt',
     component: TechRecordComponent,
-    data: { title: 'Tech Record' }
+    data: { title: 'Tech Record' },
```

Comment:
Do we need the canActivateChild guard here too?

Reply:
I am not sure it is needed on either. This route does not have a child route, I think it is redundant code.